### DISCUSSION TITLE
What is the rationale for putting save-exact into .npmrc?

### DISCUSSION THREAD

**Discussion by vincentrolfs** [blitz-js/blitz#3244] [Q&A] [answered]
What is the rationale for putting save-exact into .npmrc? I want to understand more about why this is done, since it's not at all obvious why this is a good idea.

In former times, the main argument for save-exact was trying to avoid breakage of production builds, but with package.lock/yarn.lock this argument does not seem to apply anymore.

**Comment by beerose** [comment, marked as answer]
This is a PR that introduced it: https://github.com/blitz-js/blitz/pull/500

Even with yarn.lock/package.lock etc you still may need it as `save-exact=true` makes sure that no sliding versions (with ~ or ^ ) will not be installed. 

Example: 
* you have `"some-package": "^1.2.3"` in your package.json 
* you have the same version in a lock file
* then you deleted node_modules and want to install dependencies again 
* **and** in the meantime there's a new minor version of `some-package`, e.g. 1.8.0 
* package manager will install 1.8.0 if you don't have `save-exact=true`.
